### PR TITLE
feat(ci/e2e): create a CI job to test e2e related makefile steps on merge to main

### PR DESCRIPTION
**Pull request by anhductn2001**
Closes: #3208 

## What is the purpose of the change

> Create a CI job to test e2e related makefile steps on merge to main

## Brief Changelog

  - *Run make docker-build-e2e-init-chain on merge to main*
  - *Run make test-e2e-short on merge to main*

### CODE REVIEW

**Comment by anhductn2001** [comment]
I just run in local and it worked.

File: .github/workflows/test-e2e-makefile.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: Test-E2E-Makefile
+
+on:
+  pull_request:
+    branches:
+      - "**"
```
Comment (p0mvn) [inline .github/workflows/test-e2e-makefile.yml:6]: We only want to run these on merge to `main`, not on pull requests

```suggestion
```

**Review by p0mvn — COMMENTED** [review]
Can you show me a test please similar to what we discussed here:
https://github.com/osmosis-labs/osmosis/pull/3249#issuecomment-1317491912

**Comment by anhductn2001** [comment]
Link result here: https://github.com/anhductn2001/osmosis/actions/runs/3485013832

**Review by p0mvn — APPROVED** [review]

**Pull request merged**